### PR TITLE
Add option param for observe function and unit test

### PR DESCRIPTION
**Pull request by c-a-w**

### CODE REVIEW

File: src/ViewportIntersectionObserver.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ export default class ViewportIntersectionObserver {
     this.observed = false;
     this.ignore_first_observe
       = !!(opt && opt.ignore_first_observe)
+    this.lastCheck = Date.now();
```
Comment (bonar) [inline src/ViewportIntersectionObserver.js]: The name "lastCheck" is ambiguous. Use "lastObserveTime".
And its default value should be null.

File: src/ViewportIntersectionObserver.js [hunk 2]
```diff
@@ -36,12 +37,22 @@ export default class ViewportIntersectionObserver {
     }
   }
 
-  observe() {
+  shouldUpdateTimeCheck(time) {
```
Comment (bonar) [inline src/ViewportIntersectionObserver.js]: The Function name "shouldUpdateTimeCheck" is ambiguous.
It should be "throttleTimedOut" or "isThrottleTimeout".
And its parameter name should be "interval" or "wait". ("time" may be ambiguous)

File: src/ViewportIntersectionObserver.js [hunk 3]
```diff
@@ -36,12 +37,25 @@ export default class ViewportIntersectionObserver {
...
       return;
     }
+    if (opt && opt.throttle) {
+      if (this.lastObserveTime == null) {
+        this.lastObserveTime = Date.now();
+      }
+      if (!this.throttleTimedOut(opt.throttle)) {
+        return;
+      }
+      this.lastCheck = Date.now();
```
Comment (bonar) [inline src/ViewportIntersectionObserver.js]: This should be replaced with "this.lastObserveTime".

File: test/browser/bundle.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ var ViewportIntersectionObserver = function () {
     this.registry = [];
     this.observed = false;
     this.ignore_first_observe = !!(opt && opt.ignore_first_observe);
+    this.lastCheck = Date.now();
```
Comment (bonar) [inline test/browser/bundle.js]: You need to update this file with:

```
npm run browser
```

Comment (c-a-w) [reply test/browser/bundle.js]: Sorry about that!

**Review by bonar — APPROVED** [review]
Thanks for updating code.

**Pull request merged**